tests/api/test_project.py

```diff
@@ -1079,6 +1124,326 @@ def test_projectstatus_post_invalid_deadline_expire(module_client, boto3_session
     assert "The deadline needs to be less than (or equal to) 30 days." in response.json["message"]
 
 
+def test_extend_deadline_bad_confirmed(module_client, boto3_session):
+    """Try to extend a deadline and send a not boolean for confirmation"""
+
+    # create project and release it
+    project_id, project = create_and_release_project(
+        client=module_client, proj_data=proj_data, release_data=release_project
+    )
+    assert project.times_expired == 0
+    time.sleep(1)  # tests are too fast
+
+    # try to extend deadline with a string as confirmed - should fail
+    response = module_client.patch(
+        tests.DDSEndpoint.PROJECT_STATUS,
+        headers=tests.UserAuth(tests.USER_CREDENTIALS["unitadmin"]).token(module_client),
+        query_string={"project": project_id},
+        json={**extend_deadline_data, "confirmed": "true"},
+    )
+    assert response.status_code == http.HTTPStatus.BAD_REQUEST
+    assert "`confirmed` is a boolean value: True or False." in response.json["message"]
+
+
+def test_extend_deadline_no_confirmed(module_client, boto3_session):
+    """Try to extend a deadline before confirmation - should sent a warning and no operation is perfrom"""
+
+    # create project and release it
+    project_id, project = create_and_release_project(
+        client=module_client, proj_data=proj_data, release_data=release_project
+    )
+    assert project.times_expired == 0
+    time.sleep(1)  # tests are too fast
+
+    # try to extend deadline
+    response = module_client.patch(
+        tests.DDSEndpoint.PROJECT_STATUS,
+        headers=tests.UserAuth(tests.USER_CREDENTIALS["unitadmin"]).token(module_client),
+        query_string={"project": project_id},
+        json=extend_deadline_data_no_confirmed,
+    )
+    # status code is ok but no operation perform
+    assert response.status_code == http.HTTPStatus.OK
+    assert project.times_expired == 0
+
+    assert "Operation must be confirmed before proceding." in response.json["warning"]
```

Comment:
Also verify that the project status and info etc are returned
```suggestion
    assert "Operation must be confirmed before proceding." in response.json["warning"]
    assert all(item in response.json for item in ["project_info", "project_status", "warning", "default_unit_days"])
```